src/main/webapp/app/exercises/shared/result/result.service.ts

```diff
@@ -117,4 +123,39 @@ export class ResultService implements IResultService {
         }
         return participation;
     }
+
+    /**
+     * Fetches all results for an exercise and returns them
+     */
+    getResults(exercise: Exercise) {
+        return this.getResultsForExercise(exercise.id!, {
+            withSubmissions: exercise.type === ExerciseType.MODELING,
+        }).pipe(
+            tap((res: HttpResponse<Result[]>) => {
+                return res.body!.map((result) => {
+                    result.participation!.results = [result];
+                    (result.participation! as StudentParticipation).exercise = exercise;
+                    if (result.participation!.type === ParticipationType.PROGRAMMING) {
+                        addUserIndependentRepositoryUrl(result.participation!);
+                    }
+                    result.durationInMinutes = this.durationInMinutes(
+                        result.completionDate!,
+                        result.participation!.initializationDate ? result.participation!.initializationDate : exercise.releaseDate!,
+                    );
+                    // Nest submission into participation so that it is available for the result component
+                    if (result.participation && result.submission) {
+                        result.participation.submissions = [result.submission];
+                    }
```

Comment:
```suggestion
                    if (result.submission) {
                        result.participation!.submissions = [result.submission];
                    }
```
I assume this check is not needed here since we assume the existence of `result.participation` in the aforementioned statement. If the participation is 100% there, remove the check and if it could be missing, we should probably move the participation check to the top of the subscribe. (I know this method already existed, so not your fault)

Reply:
:+1: